estate/models/estate_property_offer.py

```diff
@@ -0,0 +1,77 @@
+from odoo import api, exceptions, fields, models
+
+class EstatePropertyOffer(models.Model):
+    _name = 'estate.property.offer'
+    _description = 'Estate property offer'
+
+    active = fields.Boolean('Is Active?', default=True)
+    price = fields.Float('Price')
+    status = fields.Selection([
+            ('accepted', 'Accepted'),
+            ('refused', 'Refused'),
+        ], string='Status', readonly=True
+    )
+    partner_id = fields.Many2one('res.partner', string='Partner', required=True)
+    property_id = fields.Many2one('estate.property', string='Property', required=True)
+    property_type_id = fields.Many2one('estate.property.type',
+        string="Type",
+        related='property_id.property_type_id'
+    )
+    validity = fields.Integer('Validity (days)', default=7)
+    date_deadline = fields.Date('Deadline',
+        compute='_compute_date_deadline',
+        inverse='_inverse_date_deadline',
+    )
+
+    _order = 'price desc'
+    _sql_constraints = [
+        ('offer_price_strictly_positive', 'CHECK(price > 0)',
+        'Offer price should be stricly positive.')
+    ]
+
+    @api.depends('validity', 'create_date')
+    def _compute_date_deadline(self):
+        for record in self:
+            if record.create_date:
+                record.date_deadline = fields.Date.add(record.create_date, days=record.validity)
+            else:
+                record.date_deadline = fields.Date.add(fields.Date.today(), days=record.validity)
+
+    def _inverse_date_deadline(self):
+        for record in self:
+            record.validity = (record.date_deadline - record.create_date.date()).days
+
+    def action_accept_offer(self):
+        for record in self:
+            match record.property_id.state:
+                case 'new' | 'offer_received':
+                    for offer in record.property_id.offer_ids:
+                        offer.status = 'refused'
+                    record.property_id.buyer_id = record.partner_id
+                    record.property_id.selling_price = record.price
+                    record.property_id.state = 'offer_accepted'
+                    record.status = 'accepted'
+                case 'offer_accepted' | 'sold':
+                    raise exceptions.UserError('An offer has already been accepted')
+                case 'canceled':
+                    raise exceptions.UserError('This sale has been canceled')
+        return True
+
+    def action_refuse_offer(self):
+        for record in self:
+            match record.status:
+                case 'accepted':
+                    raise exceptions.UserError('You cannot refuse an accepted offer')
+                case False:
+                    record.status = 'refused'
+        return True
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        for vals in vals_list:
+            self.env['estate.property'].browse(vals['property_id']).state = 'offer_received'
```

Comment:
will crash if 'property_id' not in vals.
also, I would do that after the create.
so you can use the created record to access their properties